pymeasure/instruments/keysight/keysightN7776C.py

```diff
@@ -0,0 +1,239 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2021 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+import numpy as np
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+WL_RANGE = [1480, 1620]
+LOCK_PW = 1234
+
+
+class KeysightN7776C(Instrument):
+    """
+    This represents the Keysight N7776C Tunable Laser Source interface.
+
+    .. code-block:: python
+
+        laser = N7776C(address)
+        laser.sweep_wl_start = 1550
+        laser.sweep_wl_stop = 1560
+        laser.sweep_speed = 1
+        laser.sweep_mode = 'CONT'
+        laser.output_enabled = 1
+        while laser.sweep_state == 1:
+            log.info('Sweep in progress.')
+        laser.output_enabled = 0
+
+    """
+    def __init__(self, address, **kwargs):
+        super(KeysightN7776C, self).__init__(
+            address, "N7776C Tunable Laser Source", **kwargs)
+
+    locked = Instrument.control(':LOCK?', ':LOCK %g,'+str(LOCK_PW),
+                                """ Boolean property controlling the lock state (True/False) of \
```

Comment:
What's up with the line continuation `\` here? These are multiline strings, I'd suggest to leave the natural linebreaks in place. Also below. Also, you should not need to add a `\n` at the end of lines, that feels weird. 

Just write the docstrings with natural linebreaks, and if you absolutely want it to fit on one line without hitting the line length limit, you can use an indentation pattern of
```
    output_enabled = Instrument.control(
        'SOUR0:POW:STAT?', 
        'SOUR0:POW:STAT %g',
        """ Boolean Property that controls the state (on/off) of the laser source """,
        validator=strict_discrete_set,
        map_values=True,
        values={True: 1, False: 0},
    )
```

Reply:
I changed the docstrings, at one point a `\` seems to be unavoidable, or I'm to inexperienced with the linting and linebreaks. Feel free to change it if you know a better way.

```
        elif response == 374:
            log.warning('The only allowed modulation source with lambda logging \
                        function is coherence control.')
```